tools/lsp/ui/resizer.slint

```diff
@@ -0,0 +1,88 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+global ResizeState {
+    out property <length> handle-size: 10px;
+}
+
+component ResizeHandle inherits Rectangle {
+    callback resize(/* width */ length, /* height */ length);
+    in property <MouseCursor> mouse-cursor;
+
+    width: ResizeState.handle-size;
+    height: ResizeState.handle-size;
+
+    background: Colors.black;
+    border-color: Colors.white;
+    border-width: 1px;
+
+    TouchArea {
+        moved() => {
+            root.resize(self.mouse-x - self.pressed-x, self.mouse-y - self.pressed-y);
+        }
+        mouse-cursor <=> root.mouse-cursor;
+    }
+}
+
+export component Resizer {
+    in property <bool> is-resizable: true;
+    callback resize(/* width */ length, /* height */ length);
+
+    property <length> handle-size: ResizeState.handle-size;
+
+    resize-area := Rectangle {
+        width <=> parent.width;
+        height <=> parent.height;
+
+        @children
+    }
+
+    if root.is-resizable: ResizeHandle { // N
```

Comment:
Instead of having many `if` , you can combine them in one:

```
if root.is-resizable: Rectangle {
    width: 100%; height: 100%; // the default, IIRC
    ResizeHandle { ..}
    ResizeHandle { ..}
    ResizeHandle { ..}
    // ...
}
```

One empty rectangle is really cheap, but all the `if` expressions adds a bit of overhead.   Also makes the code simpler, IMHO.